libmscore/system.cpp

```diff
@@ -162,6 +162,7 @@ Box* System::vbox() const
 SysStaff* System::insertStaff(int idx)
       {
       SysStaff* staff = new SysStaff;
+      staff->idx = idx;
```

Comment:
In order to maintain `SysStaff::idx` value correctly this method should probably check whether it appends a staff to the end of staves list or inserts it somewhere in the middle and shift other indices staves correspondingly in the latter case. Something similar should probably be done in `System::removeStaff()`.

Still if the only place where this `idx` is used is here:
https://github.com/musescore/MuseScore/blob/3b9cf4d961020a62bec3aa0671bcef672edad341/libmscore/system.cpp#L1352
then maybe we could get rid of `idx` entirely and make `firstVisibleSysStaff`/`lastVisibleSysStaff` (or some similar functions) tell us the staff index directly based on its position in staves list. Then we wouldn't need to maintain `idx` value ourselves.

Reply:
Right now this list is never modified, only rebuilt from scratch, but yeah, better safe than sorry.  It's also possible, btw, to set idx during layout when it's even safer.   Somehow it does seems like there should be a way to get from a SyStaff back to the actual staff, even if right now this is the only place we seem to need it.  But, maybe not something I need to solve, changing the other functions to return the index is probably the better way to go for now.  Will do.